src/main/java/com/uber/cadence/serviceclient/WorkflowServiceTChannel.java

```diff
@@ -2297,15 +2297,46 @@ public void StartWorkflowExecution(
     throw new UnsupportedOperationException("not implemented");
   }
 
+  private Long validateAndUpdateTimeout(Long timeoutInMillis) {
+    if (timeoutInMillis <= 0 || timeoutInMillis == Long.MAX_VALUE) {
```

Comment:
if (timeoutInMillis <= 0 || timeoutInMillis > options.getRpcLongPollTimeoutMillis()) {

let's cap it at options.getRpcLongPollTimeoutMillis()

Reply:
I feel cap at cadence default settings may surprise users. Since we add the timeout explicitly in API, would it make more sense to prioritize use input?